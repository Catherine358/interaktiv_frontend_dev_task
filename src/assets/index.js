export const dataProtection = require('./images/data_protection.png');
export const fraudPrevention = require('./images/fraud_prevention.png');
export const itSecurity = require('./images/it_security.png');
export const wphg = require('./images/wphg.png');
