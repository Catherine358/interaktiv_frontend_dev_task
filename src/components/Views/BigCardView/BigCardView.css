div.ui.segment.big-view {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  flex-wrap: wrap;
}

.big-card-img-container {
    width: 35px;
    height: 35px;
    margin-right: 18px;
}

.big-card-details {
    display: flex;
    flex: 1;
    flex-direction: column;
}
