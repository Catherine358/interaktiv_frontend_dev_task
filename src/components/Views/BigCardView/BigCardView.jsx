import { Image, Segment } from 'semantic-ui-react';
import PropTypes from 'prop-types';

import './BigCardView.css';

const BigCardView = (props) => {
  const { image, category, dataTestId } = props;

  return (
    <Segment className="big-view" horizontal="true" data-testid={dataTestId}>
      <div className="big-card-img-container">
        <Image src={image} alt={category} width={35} height={35} />
      </div>
      <div className="big-card-details">
        {props.children}
      </div>
    </Segment>
  );
};

export default BigCardView;

BigCardView.propTypes = {
  image: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  dataTestId: PropTypes.string.isRequired
};

BigCardView.defaultProps = {
  image: '',
  category: '',
  dataTestId: ''
};
