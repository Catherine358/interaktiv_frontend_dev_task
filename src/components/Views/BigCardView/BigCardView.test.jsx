import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { dataProtection } from '../../../assets';

import BigCardView from './BigCardView';

describe('BigCardView', () => {
  it('renders a big card view item', () => {
    const component = render(
      <BigCardView image={dataProtection} category="Data Protection"  />
    );
    expect(component).toMatchSnapshot();
    const image = component.getByAltText('Data Protection');
    expect(image).toBeInTheDocument();
  });
});
