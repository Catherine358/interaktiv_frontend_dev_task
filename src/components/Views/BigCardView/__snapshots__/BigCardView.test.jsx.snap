// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigCardView renders a big card view item 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui segment big-view"
        data-testid=""
        horizontal="true"
      >
        <div
          class="big-card-img-container"
        >
          <img
            alt="Data Protection"
            class="ui image"
            height="35"
            src="data_protection.png"
            width="35"
          />
        </div>
        <div
          class="big-card-details"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui segment big-view"
      data-testid=""
      horizontal="true"
    >
      <div
        class="big-card-img-container"
      >
        <img
          alt="Data Protection"
          class="ui image"
          height="35"
          src="data_protection.png"
          width="35"
        />
      </div>
      <div
        class="big-card-details"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
